pkg/database/batch.go

```diff
@@ -43,7 +46,10 @@ func NewBatchWithRetry(ctx context.Context, dao *DAO, syncResponse *model.SyncRe
 }
 
 // Adds a query to the queue and check if there's enough items to process the batch.
-func (b *batchWithRetry) Queue(item batchItem) {
+func (b *batchWithRetry) Queue(item batchItem) error {
+	if b.connError != nil { // Can't queue more items after DB connection error.
+		return b.connError
```

Comment:
Sine we don't check error after each item, will it keep on queuing stuff even after conn error?

Reply:
Ok, So once it is set, we will keep on returning the err and not append further items.

Reply:
That's correct.
Maybe I would have done it differently if implementing from scratch, but here I was trying to keep changes to minimum.